src/main/java/org/jabref/model/strings/StringUtil.java

```diff
@@ -167,6 +168,83 @@ public static String getCorrectFileName(String orgName, String defaultExtension)
         return orgName;
     }
 
+    /**
+     * Shorten a given file name in the middle of the name using ellipsis. Example: verylongfilenameisthis.pdf
+     * with maxLength = 20 is shortened into verylo...isthis.pdf
+     *
+     * @param fileName  the given file name to be shortened
+     * @param maxLength the maximum number of characters in the string after shortening (including the extension)
+     * @return the original fileName if fileName.length() <= maxLength. Otherwise, a shortened fileName
+     */
+    public static String shortenFileName(String fileName, int maxLength) {
+        if (fileName == null || maxLength < 3) {
+            return "";
+        }
+
+        if (fileName.length() <= maxLength) {
+            return fileName;
+        }
+
+        // Reference the ellipsis' length for future usage
+        final int ELLIPSIS_LENGTH = ELLIPSIS.length();
+
+        // Find the index of the extension dot
+        int extensionIndex = fileName.lastIndexOf('.');
+
+        // Extract the name and extension part of a file name
+        String name;
+        String extension;
+
+        // Check if file has an extension or not
```

Comment:
Please refactor to use `org.jabref.logic.util.io.FileUtil#getFileExtension(...)`

It is OK for me to use `FileUtil.getFileExtension(fileName).map(extension -> "." + extension).orElse("")`.